### ISSUE TITLE
[Feature request] Including a Furnace into the BackPacks

### ISSUE THREAD

**Issue by VincentLongiug** [Tiviacz1337/Travelers-Backpack#56]
As you stated on CurseForge, having a Frunace included in Backpacks is the last missing essential thing for Travellers.
If I knew a thing about coding, I would have pulled a request for that to save you from some hard work...
Instead, I'm only able to create this issue, provide a GUI texture and try to explain my thoughts with a kind of algorithm :

#### What will we see ?
On the image below, I tried to fit a Furnace space (Gray Slots) in the BackPack, so I had to move the Tool Slots (it's not visible here, but the SleepBag/Put on-Take off button will need to be moved upper either).
*By the way, it may be nice to dissociate them when the BagPack is placed on the ground : the Put on button on the left and the Sleep Bag button on the right... Off-topic !*

The slot on the left will contain the Result, the one on the bottom right the Ingredient and the top right slot is for the Fuel.
The Fuel Slot will stand for the remaining burning time instead of the classic Fire Jauge by becoming red (thanks to the Red Slot below the GUI) when a Fuel Item is consumed and turning back to gray gradually from top to bottom.
The bar below the Tank stands for the Time Passed of the Recipe and will be filled from right to left with the texture below the Red Slot.

*I hope it's possible to do it like this : my dread is that the Red Slot overlays completely the Items in the Fuel Slot while I want it to stay beneath them. If it happens, just ask me for a more conventionnal Fire Jauge.*

![travellers_backpack](https://user-images.githubusercontent.com/52882424/62692989-e9017100-b9d1-11e9-9989-0b6cba6d7304.png) and the empty_fuel_slot.png (to avoid confusion) : ![empty_fuel_slot](https://user-images.githubusercontent.com/52882424/62694808-5236b380-b9d5-11e9-86d4-25e21176d577.png)
#### How will it works ?
Obviously, the Furnace space can hold Items as well as the Crafting space does (you can't put Items in the Result Slot direclty though), but beware of what you put in them, because the Furnace will automaticly work if it can : since Lava Bucket can be used to cook 100 Items, each 10mB of Lava in your BackPack Tanks will be used to cook 1 Item if possible.
```
if (Item in Ingredient Slot is "cookable"), check Item in Fuel Slot
   if (Item in Fuel Slot is fuel), consume 1 Item in Fuel Slot    //as a normal Furnace would do
   else check BackPack Tanks    //at this state, the Fuel Slot is either empty or filled with non-fuel Items
      if (BackPack Tanks contains 10mB of Lava), use 10mB of Lava to cook 1 Item
      else do nothing
else do nothing
```
*I didn't use the correct coding synthax on purpose*

**Comment by DWX12** [comment]
The backpack does alot already. As someone who has used the mod since 1.7.10 (well the old version of it), I don't feel that the backpack should have a furnace inside it. The backpack shouldn't be a one stop shop for everything. I even feel that the crafting table is a bit much as is, but it's not something I would petition on removing. But this is just from a player's perspective.

It's up to Tiviacz to decide on if he wants to add this idea into the mod. At the moment, it's mainly just a port of the original.

**Comment by Tiviacz1337** [comment]
Well, that's a really nice idea, but as for now, I'll not work on it

**Comment by shaneknysh** [comment]
I think adding a furnace to the exiting backpack makes an already powerful tool pushing into OP. I'd prefer to have the crafting table replaces by the the furnace, blast furnace, smoker, etc

that would allow you to pick which one you need. 

Or if you are going to have them added then having the furnace take up an additional 9 storage blocks so there is a cost to adding the additional functionality.

**Comment by shaneknysh** [comment]
If a furnace is added to the backpack the ability to use lava from a tank - possibly by changing the hose state.

**Comment by WubTheGame** [comment]
I don't disagree with putting a furnace into it, but at that point if you ask me it would be a solid idea to have it be an upgrade like others have suggested, and/or add a toggle for the furnace and even crafting table to have or not have them entirely. Let the player decide how "balanced" they want this to be.

**Comment by GordonRamsay20** [comment]
They do realise you can just put a furnace in your backpack right

**Comment by shaneknysh** [comment]
They also realize you have to place the furnace down to use it, making it slightly less mobile than the crafting table in the backpack.

**Comment by GordonRamsay20** [comment]
> They also realize you have to place the furnace down to use it, making it slightly less mobile than the crafting table in the backpack.

Oh no! One or two extra clicks, it's the end of the world!

**Comment by shaneknysh** [comment]
Yes, a similar number of saved clicks as say using this mod vs carrying around a shulker box with a crafting table, 4 buckets of water, 4 buckets of lava, and a bed.

Minecraft is a big world. Other people might want things you you don't want. Might play differently than you. If you are counting clicks and passing judgement you can save a lot of time and extra clicks by playing vanilla.

**Comment by Aceplante2** [comment]
> I think adding a furnace to the exiting backpack makes an already powerful tool pushing into OP. I'd prefer to have the crafting table replaces by the the furnace, blast furnace, smoker, etc
> that would allow you to pick which one you need.
> Or if you are going to have them added then having the furnace take up an additional 9 storage blocks so there is a cost to adding the additional functionality.

Yes, I like the idea of having it be an upgrade you can do INSTEAD of the crafting table. So you may want to put a furnace upgrade in your mining pack, but a crafting table in your exploring pack etc.


> I don't disagree with putting a furnace into it, but at that point if you ask me it would be a solid idea to have it be an upgrade like others have suggested, and/or add a toggle for the furnace and even crafting table to have or not have them entirely. Let the player decide how "balanced" they want this to be.

Yes exactly! In fact, I just suggested a way to do just that lol

**Comment by Tiviacz1337** [comment]
https://www.curseforge.com/minecraft/mc-mods/travelers-backpack

Furnace, Smoker and Blast Furnace upgrades available for Forge & Fabric 1.20.1
One of the oldest suggestions, took me a while to gain knowledge and experience to add such feature, but it's finally here :P

**Comment by Tiviacz1337** [comment]
https://www.curseforge.com/minecraft/mc-mods/travelers-backpack

Upgrades available for all other versions: 1.21.1+ Neo and Fabric
Closing
